test/connection-checker/src/net.rs

```diff
@@ -74,3 +75,53 @@ pub fn send_ping(opt: &Opt, destination: IpAddr) -> eyre::Result<()> {
 
     Ok(())
 }
+
+#[cfg(target_os = "macos")]
+pub fn send_ping(opt: &Opt, destination: IpAddr) -> eyre::Result<()> {
+    eprintln!("Leaking ICMP packets to {destination}");
+
+    ping::dgramsock::ping(
+        destination,
+        Some(Duration::from_millis(opt.leak_timeout)),
+        None,
+        None,
+        None,
+        None,
+    )?;
+
+    Ok(())
+}
+
+// Some Linux distributions don't allow unprivileged users to send ICMP packets.
+// We use the ping command (which has capabilities/setuid set) to get around that.
+#[cfg(target_os = "linux")]
+pub fn send_ping(opt: &Opt, destination: IpAddr) -> eyre::Result<()> {
+    eprintln!("Leaking ICMP packets to {destination}");
+
+    let mut cmd = std::process::Command::new("ping");
+
+    // NOTE: Rounding up to nearest second, since some versions don't support fractional
+    //       seconds
+    let timeout_sec = ((opt.leak_timeout + 1000 - 1) / 1000).to_string();
```

Comment:
`ConnCheckerHandle::check_connection` sort of relies on the timing here. Are we sure this won't cause a bug? If there's doubt, maybe we should just change the type of `Opt::leak_timeout` to seconds as well.

Reply:
Changed to seconds. :)